client/http/client.go

```diff
@@ -190,6 +192,12 @@ func (ci *clientInner) doRequest(
 		opt(req.Header)
 	}
 	req.Header.Set(xCallerIDKey, callerID)
+	if req.Header.Get(XForbiddenForwardToMicroServiceHeader) == "true" {
+		if req.Header.Get(XForwardedToMicroServiceHeader) == "true" {
```

Comment:
```suggestion
	if req.Header.Get(XForbiddenForwardToMicroServiceHeader) == "true" && req.Header.Get(XForwardedToMicroServiceHeader) == "true" {
```

Reply:
And what about wrapping a function to check and set up the header?

Reply:
checked in https://github.com/tikv/pd/pull/7834/files#diff-138f7d23a1171f829bb57a7edcd66e8ac0ceb81f1cb2f28e497284c732aaad22R374